app/shared/hooks/useSearchFilters.js

```diff
@@ -0,0 +1,40 @@
+'use client'
+import { useContext, useMemo } from 'react'
+import { SearchFiltersContext } from './SearchFiltersProvider'
+
+export const useSearchFilters = (issues = []) => {
+  const context = useContext(SearchFiltersContext)
+  if (!context) {
+    throw new Error(
+      'useSearchFilters must be used within a SearchFiltersProvider',
+    )
+  }
+
+  const [filters, setFilters] = context
+
+  const filteredIssues = useMemo(() => {
+    return (
+      issues?.filter((issue) => {
+        // Search filter
+        const matchesSearch =
+          filters.search === '' ||
+          issue.title?.toLowerCase().includes(filters.search.toLowerCase()) ||
+          issue.description
+            ?.toLowerCase()
+            .includes(filters.search.toLowerCase())
+
+        // Status filter
+        const matchesStatus =
+          filters.status === 'all' || issue.status === filters.status
+
+        return matchesSearch && matchesStatus
+      }) || []
+    )
+  }, [issues, filters])
+
+  return {
+    issues: filteredIssues,
+    filters,
+    setFilters,
+  }
```

Comment:
```suggestion
  return [ filters, setFilters, filterItems ]
```

If we just expost the filters and the method to filter the items by the filters, then we limit dependencies here.

the consumer can then just do something like:

```javascript
const [items] = useItems(initItems)
const [filters, setFilters, filterItems] = useSearchFilters({ status: 'all', search: '' })
const [filteredItems, setFilteredItems] = useState(items)

useEffect(() => setFilteredItems(filterItems(items)), [filters])

const onSearchChange = (search = '') => 
  setFilters({...filters, search})
```